adding uml class diagram of example_1 (#914)
ros-controls/ros2_control_demos: example_1/doc/userdoc.rst

.. include:: ../../doc/run_from_docker.rst

RRBot Architecture Overview
---------------------------

.. image:: ros2_control_demo_example_1.png
   :width: 600
   :align: center
   :alt: ROS 2 Control Demo Example 1 Architecture Overview

This section provides an overview of how **ROS2 Control** components interact in *Example 1 (RRBot System Position Only)*. It summarizes the relationship between the hardware interface, controllers, and the robot’s URDF.

A custom hardware interface ``RRBotSystemPositionOnlyHardware`` is implemented in:
``ros2_control_demo_example_1/rrbot_hardware_interface.cpp``

This hardware plugin:

- Inherits from ``hardware_interface::SystemInterface``
- Defines how to **read joint positions** (from hardware, simulated here)
- Defines how to **write commands** (set joint positions)
- Is registered in a ``plugin.xml`` file so the **ResourceManager** can load it

From the UML:

- ``hardware_interface::SystemInterface`` → implemented by the RRBot hardware plugin

> - ``hardware_interface::ResourceManager`` → manages your hardware plugin after loading it

Reviewer: ```suggestion
- ``hardware_interface::ResourceManager`` → loads and manages the hardware plugins
```
